### ISSUE TITLE
File rename refactoring also renames constructor

### ISSUE THREAD

**Issue by mayakwd** [HaxeFoundation/intellij-haxe#785]
Actual behavior:
```haxe
class Test {
  public function new() {
  }
}
```

After rename:

```haxe
class RenamedTest {
  public function RenamedTest() {
  }
}
```

Expected result: constructor's name remaining the same

**Comment by mayakwd** [comment]
During investigation I discovered that it's not possible to fix issue by implement custom `RenameProcessor`. It happens because all compatible rename processors are triggered in the rename chain. RenameJavaClassProcessor is compatible as well because the only guard that it uses: "PsiElement should be instance of PsiClass", and it renaming the class constructor. The only possible solution is to rename constructor backward after it was renamed.

**Comment by EricBishton** [comment]
Also see #292, #776

**Comment by EricBishton** [comment]
To be clear: Refactor->Rename File... does NOT change the name of the constructor, but it does change the name of the class (correctly).  Refactor->Rename... (Shift+F6), which is a more common operation, exhibits the bug.

Verified on 1.1(+develop).

**Comment by EricBishton** [comment]
Fixed #922
